tests/components/qwikswitch/test_init.py

```diff
@@ -106,3 +120,299 @@ async def test_sensor_device(hass, aioclient_mock):
+    )
+    await asyncio.sleep(0.01)
+    assert (
+        "GET",
+        URL("http://127.0.0.1:2020/@a00003=0"),
+        None,
+        None,
+    ) in aioclient_mock.mock_calls
+    state_obj = hass.states.get("light.dim_3")
+    assert state_obj.state == "off"
+
+    # change brightness in network and check that hass updates
+    qs_devices[2]["val"] = "280c55"  # half dimmed
+    listen_mock.queue_response(json=EMPTY_PACKET)
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    state_obj = hass.states.get("light.dim_3")
+    assert state_obj.state == "on"
+    assert 16 < state_obj.attributes["brightness"] < 240
+
+    # turn off in the network and see that it is off in hass as well
+    qs_devices[2]["val"] = "280c78"  # off
+    listen_mock.queue_response(json=EMPTY_PACKET)
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    state_obj = hass.states.get("light.dim_3")
+    assert state_obj.state == "off"
+
+    # ask hass to turn on and verify command is sent to device
+    aioclient_mock.mock_calls.clear()
+    aioclient_mock.get("http://127.0.0.1:2020/@a00003=100", json={"data": "OK"})
+    await hass.services.async_call(
+        "light", "turn_on", {"entity_id": "light.dim_3"}, blocking=True
+    )
+    assert (
+        "GET",
+        URL("http://127.0.0.1:2020/@a00003=100"),
+        None,
+        None,
+    ) in aioclient_mock.mock_calls
+    await hass.async_block_till_done()
+    state_obj = hass.states.get("light.dim_3")
+    assert state_obj.state == "on"
+
+    listen_mock.stop()
+
+
+async def test_button(hass, aioclient_mock, qs_devices):
+    """Test that buttons fire an event."""
+
+    async def get_devices_json(method, url, data):
+        return AiohttpClientMockResponse(method=method, url=url, json=qs_devices)
+
+    config = {"qwikswitch": {"button_events": "TOGGLE"}}
+    aioclient_mock.get("http://127.0.0.1:2020/&device", side_effect=get_devices_json)
+    listen_mock = MockLongPollSideEffect()
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", side_effect=listen_mock)
+    assert await async_setup_component(hass, QWIKSWITCH, config)
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    button_pressed = Mock()
+    hass.bus.async_listen_once("qwikswitch.button.@a00002", button_pressed)
+    listen_mock.queue_response(json={"id": "@a00002", "cmd": "TOGGLE"},)
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    button_pressed.assert_called_once()
+
+    listen_mock.stop()
+
+
+async def test_failed_update_devices(hass, aioclient_mock):
+    """Test that code behaves correctly when unable to get the devices."""
+
+    config = {"qwikswitch": {}}
+    aioclient_mock.get("http://127.0.0.1:2020/&device", exc=ClientError())
+    listen_mock = MockLongPollSideEffect()
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", side_effect=listen_mock)
+    assert not await async_setup_component(hass, QWIKSWITCH, config)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5403678)